test/e2e-skipper-tests.sh

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+# This script runs e2e tests for Canary initialization, analysis and promotion
+# Prerequisites: Kubernetes Kind and Skipper ingress controller
+
+set -o errexit
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+
+echo '>>> Creating test namespace'
+kubectl create namespace test
+
+echo '>>> Installing load tester'
+kubectl apply -k ${REPO_ROOT}/kustomize/tester
+kubectl -n test rollout status deployment/flagger-loadtester
+
+echo '>>> Initialising canary'
+kubectl apply -f ${REPO_ROOT}/test/e2e-workload.yaml
+kubectl apply -f ${REPO_ROOT}/test/e2e-ingress-skipper.yaml
+
+echo '>>> Create canary CRD'
+cat <<EOF | kubectl apply -f -
+apiVersion: flagger.app/v1beta1
+kind: Canary
+metadata:
+  name: podinfo
+  namespace: test
+spec:
```

Comment:
think the optional provider is missing: should be `skipper`  
https://github.com/weaveworks/flagger#canary-crd

got with skipper provider an error (which was expected):

```
{"level":"info","ts":"2020-08-04T13:26:11.961Z","caller":"canary/deployment_controller.go:282","msg":"Deployment podinfo-primary.test created","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:11.965Z","caller":"controller/events.go:28","msg":"podinfo-primary.test not ready: waiting for rollout to finish: observed deployment generation less then desired generation","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:26.841Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:26.860Z","caller":"controller/events.go:28","msg":"podinfo-primary.test not ready: waiting for rollout to finish: 0 of 1 updated replicas are available","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:41.827Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:41.833Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:41.916Z","caller":"router/kubernetes_default.go:157","msg":"Service podinfo.test created","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:41.917Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:56.827Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:56.842Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:26:56.848Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:11.842Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:11.854Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:11.869Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:26.842Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:26.856Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:26.865Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:41.842Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:41.854Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:41.866Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:56.841Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:56.858Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:27:56.873Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:11.839Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:11.869Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:11.883Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:26.842Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:26.872Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:26.887Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:41.832Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:41.848Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:41.852Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:56.840Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:56.852Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:28:56.866Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:11.840Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:11.854Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:11.869Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:26.839Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:26.856Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:26.871Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:41.847Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:41.857Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:41.873Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:56.840Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:56.850Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:29:56.864Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:30:11.843Z","caller":"controller/events.go:16","msg":"all the metrics providers are available!","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:30:11.853Z","caller":"canary/deployment_controller.go:46","msg":"Scaling down Deployment podinfo.test","canary":"podinfo.test"}
{"level":"info","ts":"2020-08-04T13:30:11.869Z","caller":"controller/events.go:28","msg":"reconcileDestinationRule failed: DestinationRule podinfo-canary.test create error: the server could not find the requested resource (post destinationrules.networking.istio.io)","canary":"podinfo.test"}
No more retries left
```

Reply:
that explains why we get the Istio messages:

https://github.com/weaveworks/flagger/blob/master/pkg/router/factory.go#L127

Reply:
ammended to the commit